src/contact.rs

```diff
@@ -869,47 +869,46 @@ impl Contact {
         Ok(ret)
     }
 
-    // add blocked mailinglists as contacts
-    // to allow unblocking them as if they are contacts
-    // (this way, only one unblock-ffi is needed and only one set of ui-functions,
-    // from the users perspective,
-    // there is not much difference in an email- and a mailinglist-address)
+    /// Adds blocked mailinglists as contacts
+    /// to allow unblocking them as if they are contacts
+    /// (this way, only one unblock-ffi is needed and only one set of ui-functions,
+    /// from the users perspective,
+    /// there is not much difference in an email- and a mailinglist-address)
     async fn update_blocked_mailinglist_contacts(context: &Context) -> Result<()> {
-        let blocked_mailinglists = context
+        context
             .sql
-            .query_map(
-                "SELECT name, grpid FROM chats WHERE type=? AND blocked=?;",
-                paramsv![Chattype::Mailinglist, Blocked::Yes],
-                |row| Ok((row.get::<_, String>(0)?, row.get::<_, String>(1)?)),
-                |rows| {
-                    rows.collect::<std::result::Result<Vec<_>, _>>()
-                        .map_err(Into::into)
-                },
-            )
+            .transaction(move |transaction| {
+                let mut stmt = transaction
+                    .prepare("SELECT name, grpid FROM chats WHERE type=? AND blocked=?")?;
+                let rows =
+                    stmt.query_map(params![Chattype::Mailinglist, Blocked::Yes], |row| {
```

Comment:
just out of interest as i ran into similar things at some point: `stmt` is needed here as the tooling for the transaction-object lacks a `query_map("SELECT ...", ...)` as available on the sql-object? same below for `excute()` instead of `exist()`?

as said, just asking, fine for me, not suggesting to extend transaction-object or so :)